packages/compiler-cli/src/ngtsc/reflection/src/type_to_value.ts

```diff
@@ -38,7 +38,11 @@ export function typeToValue(
   // has a value declaration associated with it. Note that const enums are an exception,
   // because while they do have a value declaration, they don't exist at runtime.
   if (decl.valueDeclaration === undefined || decl.flags & ts.SymbolFlags.ConstEnum) {
-    return noValueDeclaration(typeNode, decl.declarations[0]);
+    let typeOnlyDecl: ts.Declaration|null = null;
+    if (decl.declarations !== undefined && decl.declarations.length > 0) {
+      typeOnlyDecl = decl.declarations[0];
+    }
```

Comment:
Any reason not to use a ternary operator and a `const` here?

```suggestion
    const typeOnlyDecl = (decl.declarations !== undefined && decl.declarations.length > 0) ? decl.declarations[0] : null;
```

or even cooler:

```suggestion
    const typeOnlyDecl = decl.declarations?.[0] ?? null;
```

Reply:
Because @alxhub is avoiding the implicit undefined checks (although there is one right in this file at line 53).

I did have something similar to your second, 🆒er proposal (but I didn't use the `?.[0]`, that was too 🆒 for me) but then switched over to the more explicit form.

I don't mind either way, really :-)

Reply:
😆 

I do find `decl.declarations?.[0] ?? null` just a little too concise and tricky for my brain to parse. Maybe I'm just not wired for safe property navigation

Reply:
Give it time @alxhub ⏳ 😸